### DISCUSSION TITLE
Missing HmIP-RGBW

### DISCUSSION THREAD

**Discussion by Boarder** [SukramJ/aiohomematic#1031] [Ideas]
### The problem

The new Controler HmIP-RGBW is not showing up as a light entity.
I didn´t find any why to build a dashboard with a light switch.
<img width="1350" alt="HmIP-RGBW" src="https://user-images.githubusercontent.com/45979973/232503090-4f33eb65-9777-4e17-bbe9-73495fe52882.png">

**Comment by SukramJ** [comment]
This device is currently not supported. Please create device descriptions as described [here](https://github.com/danielperna84/custom_homematic#homematicip_localexport_device_definition) and add them here too.
Please add some screenshots from ccu web ui.

**Comment by Boarder** [comment]
LED Controller - RGBW

In every situation at home the right lighting mood with LED strips: Simply set up with the LED Controller - RGBW from Homematic IP.
![image](https://user-images.githubusercontent.com/45979973/233633304-b68cddab-a2fd-408b-84d1-a57148a3a0dc.png)

![image](https://user-images.githubusercontent.com/45979973/233633443-96f7c6af-d5c4-4aa2-955b-4be3e7177149.png)

**Reply by danielperna84** [reply]
You also have to export the device definition, as mentioned above, and provide it here. It tells us the names of the parameters. Without them it's not possible to add support.

**Reply by 011V32** [reply]
I have the same problem. Here are my files
[HmIP-RGBW.zip](https://github.com/danielperna84/hahomematic/files/11320962/HmIP-RGBW.zip)

**Reply by 011V32** [reply]
But maybe it makes different than what the device is configured?
![image](https://user-images.githubusercontent.com/22346481/234234569-dcf44f0d-2c1c-4fe3-98b0-fc767f8c882a.png)

**Reply by SukramJ** [reply]
Version 1.35.0b0 unterstützt jetzt den HmIP-RGBW zumindest in den Grundfunktionen.
Bitte beim Testen auch mal die verschiedenen Modi ausprobieren. Ein Neustart von HA ist danach aber erforderlich.

**Reply by Boarder** [reply]
Was muss ich tun, damit ich die Version 1.35.0b0 installieren kann?

**Reply by SukramJ** [reply]
Die Beta in HACS aktivieren:
Integrationen --> Homematic(IP) Local --> drei Punkte(oben rechts) --> erneut herunterladen --> Beta aktivieren

**Reply by 011V32** [reply]
Ich bekomme keine Verbindung zum Gerät.
(In RaspberryMatic lässt es sich bedienen.)
![image](https://user-images.githubusercontent.com/22346481/235068546-7e9ee924-035a-4237-b2ca-b0f0af5515ab.png)

**Reply by SukramJ** [reply]
Die alten Entitäten werden natürlich nicht mehr gebraucht und können gelöscht werden.
Neu sollten natürlich light Entitäten erscheinen.
Am Besten das gerät noch mal mit dem HA Service` Homematic(IP) Local: Delete a device` aus HA löschen und HA neu starten.

**Reply by 011V32** [reply]
Habe es mit `Homematic(IP) Local: Delete a device` versucht, hat leider auch nicht funktioniert.
![image](https://user-images.githubusercontent.com/22346481/235072002-2823ff2f-b696-4e6e-9baa-f45c48d2cad4.png)

**Reply by SukramJ** [reply]
Schaut bitte auch immer ins HA Log, wenn etwas nicht funktioniert. Screenshots sind da meistens nicht aussagekräftig.
Wenn die light Entitäten initial nicht erscheinen muss ggf. der HA Service `Homematic(IP) Local: Clear the cache` ausgeführt werden. Danach HA neustarten.

**Reply by SukramJ** [reply]
@011V32 Wo ist das Problem? Wie in der Doku beschrieben sind die meisten zusätzlichen Entitäten deaktiviert.
Also must du bei Bedarf `aktivieren`.

**Reply by Boarder** [reply]
Ich bekomme es auch nicht ans laufen

**Reply by SukramJ** [reply]
@Boarder Damit kann ich nicht anfangen.
**Auch** ist hier falsch, denn wie man im Screenshot von @011V32 werden die Entitäten richtig angelegt.

**Reply by 011V32** [reply]
Ich habe die light Entität. Diese zeigt mir an, dass es eingeschalten ist obwohl es ausgeschalten ist. Wenn ich sie ausschalten möchte, wird der Schalter kurz auf aus gemacht (das Licht bleibt an) und dann geht er selber wieder auf an (obwohl es aus ist und mit dem Licht passiert nichts).
Ich habe eine Warnung welche ich vorher nicht hatte:
![image](https://user-images.githubusercontent.com/22346481/235073400-d7e8b053-7707-4a4a-b866-8d88c42215d9.png)
Die anderen Homematic Geräte funktionieren jedoch...

**Reply by SukramJ** [reply]
Aktuell wird die Entität für Kanal 1 noch nicht angezeigt. Die wird bei einer Folgeversion nachgeliefert.

**Reply by 011V32** [reply]
Bedeutet das, dass es noch gar nicht funktionieren kann?

**Reply by SukramJ** [reply]
Wie soll ich dir die Frage beantworten, wenn ich das Gerät nicht bei mir zu Hause habe?
Die Vermutung liegt aber nah, das bei einer RGBW/ RGB Konfiguration nur über Kanal 1 gesteuert wird.
Also am Besten die nächste Beta abwarten.

**Reply by 011V32** [reply]
OK. Danke. Ich kenne mich damit leider zu wenig aus. Entschuldige für die "Anfängerfragen".
Wenn ich wo etwas helfen kann (als verlängerter Arm), gerne ;)
<img width="1012" alt="image" src="https://user-images.githubusercontent.com/22346481/235080306-07632e0e-9b9d-4b96-88c0-db222085bab9.png">

**Reply by SukramJ** [reply]
In welchem Modus wird der Controller betrieben? RGB?

Könntest du mal den Modus wechseln und dann den gleichen Screenshot noch mal machen?

**Reply by SukramJ** [reply]
Der fehlende Kanal ist jetzt mit Version 1.35.0b1 verfügbar. Bitte noch mal testen.

**Reply by 011V32** [reply]
Ich bekomme folgenden Fehler:

> Logger: homeassistant.components.websocket_api.http.connection
> Source: custom_components/homematicip_local/light.py:199
> Integration: Home Assistant WebSocket API (documentation, issues)
> First occurred: 11:08:56 (4 occurrences)
> Last logged: 11:10:05
> 
> [547024062208] division by zero
> [547513229008] division by zero
> Traceback (most recent call last):
>   File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 199, in handle_call_service
>     await hass.services.async_call(
>   File "/usr/src/homeassistant/homeassistant/core.py", line 1820, in async_call
>     task.result()
>   File "/usr/src/homeassistant/homeassistant/core.py", line 1857, in _execute_service
>     await cast(Callable[[ServiceCall], Awaitable[None]], handler.job.target)(
>   File "/usr/src/homeassistant/homeassistant/helpers/entity_component.py", line 216, in handle_service
>     await service.entity_service_call(
>   File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 798, in entity_service_call
>     future.result()  # pop exception if have
>   File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 977, in async_request_call
>     await coro
>   File "/usr/src/homeassistant/homeassistant/helpers/service.py", line 838, in _handle_entity_call
>     await result
>   File "/usr/src/homeassistant/homeassistant/components/light/__init__.py", line 582, in async_handle_light_on_service
>     await light.async_turn_on(**filter_turn_on_params(light, params))
>   File "/config/custom_components/homematicip_local/light.py", line 227, in async_turn_on
>     if color_temp := kwargs.get(ATTR_COLOR_TEMP, self.color_temp):
>   File "/config/custom_components/homematicip_local/light.py", line 199, in color_temp
>     return self._hm_entity.color_temp
>   File "/usr/local/lib/python3.10/site-packages/hahomematic/platforms/support.py", line 72, in __get__
>     return self.fget(__obj)
>   File "/usr/local/lib/python3.10/site-packages/hahomematic/platforms/custom/light.py", line 456, in color_temp
>     return math.floor(1000000 / self._e_color_temperature_kelvin.value)
> ZeroDivisionError: division by zero

**Reply by SukramJ** [reply]
> In welchem Modus wird der Controller betrieben? RGB?

??

**Reply by 011V32** [reply]
RGBW

**Reply by SukramJ** [reply]
> division by zero

Das ist ein Fehler im Code, aber lass mal versuchen den zu umgehen:
Ändere mal bitte aus HA heraus die Farbtemperatur. Und teste dann noch mal.

**Reply by 011V32** [reply]
Das scheint dann zu funktionieren. 👍

Screenshot von RGBW ist oben schon.
Das hier ist RGB (sieht gleich aus wie RGBW)
<img width="951" alt="image" src="https://user-images.githubusercontent.com/22346481/235153636-150f395c-4240-4ab1-ba57-fd90726fd393.png">

Das hier ist Tunable White:
<img width="951" alt="image" src="https://user-images.githubusercontent.com/22346481/235153686-315f73e9-2684-48f9-81f7-25217c69a516.png">

Und das hier Einzelkanäle:
<img width="958" alt="image" src="https://user-images.githubusercontent.com/22346481/235153908-284419ab-7452-4f67-8294-386f55e5229e.png">

**Reply by SukramJ** [reply]
Sehr aufschlussreich. Mich wundert, das man bei RGBW im unterschied zu RGB nicht auch noch die Farbtemperatur einstellen kann.
Mir sind die Unterschiede aber auch nicht klar.
Wenn du die Farbtemperatur in HA änderst, ändert sich dann auch was an der Farbe am Gerät?

**Reply by 011V32** [reply]
Vielleicht ist nur der Unterschied, dass er den 4ten Kanal zusätzlich ansteuert um die eingestellten Farben zu erreichen.
Die Temperatureinstellung hat keine Auswirkung auf das Licht. Sollte es? (Gibt es das in Homematic?)

**Reply by SukramJ** [reply]
Dann ist die Farbtemperatur wohl nur für `Tunable White` relevant. 
Ich werde die bei RGBW wieder rausnehmen

**Reply by SukramJ** [reply]
Ist in der 1.35.0 jetzt so umgesetzt.

**Reply by 011V32** [reply]
Funktioniert super. Danke für dein Engagement. 👍

**Comment by Boarder** [comment]
![image](https://user-images.githubusercontent.com/45979973/235073526-b170e70d-7c21-4601-9cc1-9621a49b1cee.png)

**Reply by SukramJ** [reply]
Bitte auf `Vollständige Protokolldatei laden` klicken, und log als Datei anhängen
